// Copyright 2023 The Lynx Authors. All rights reserved.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package xla.auto_reorder;

import "google/protobuf/any.proto";
// Size(dim=3,sizes=[2,3,4])
message Size {
    int64 dim = 1;
    repeated int64 sizes = 2;
}
//ReplicaGroup(replica_group_id=1, replica_ids={1,2})
message ReplicaGroup{
  int64 replica_group_id=1;
  repeated int64 replica_ids=2;
}
// as xla/service/hlo.proto HloInstructionProto subset,we focus on compute/communicate complexity
message InstrProfileInfo {
  string name = 1;
  uint32 operand_count=2;
  uint32 opcode=3;
  string opname=13;
  uint32 version=4;
  repeated string operand_types = 5;
  repeated string result_types = 6;
  repeated Size operand_sizes = 7;
  Size result_size = 8;
  repeated ReplicaGroup process_groups=9;
  uint32 replica_group_size=10;
  optional string custom_call_target = 11;
  double cost=12;
  string hwinfo=14;
  string operand_hash=15;
}